### PR TITLE
add exclude config validate in compressor

### PR DESCRIPTION
**Pull request by J-shang**

### CODE REVIEW

**Comment by zheng-ningxin** [comment]
Do we have a documentation to describe the config interface of the pruner? Shall we give an example/unit test?

File: nni/algorithms/compression/pytorch/pruning/iterative_pruner.py [hunk 1]
```diff
@@ -3,8 +3,9 @@
 
 import logging
 import copy
+from nni.utils import OptimizeMode
```
Comment (zheng-ningxin) [inline nni/algorithms/compression/pytorch/pruning/iterative_pruner.py:6]: What is `OptimizeMode` for? seems not used?

File: nni/algorithms/compression/pytorch/pruning/sensitivity_pruner.py [hunk 2]
```diff
@@ -146,18 +146,23 @@ def validate_config(self, model, config_list):
...
+                Optional('sparsity'): And(float, lambda n: 0 < n < 1),
                 'op_types': ['Conv2d'],
-                Optional('op_names'): [str]
+                Optional('op_names'): [str],
+                Optional('exclude'): bool
             }], model, _logger)
 
         schema.validate(config_list)
+        for config in config_list:
+            if 'exclude' not in config and 'sparsity' not in config:
```
Comment (zheng-ningxin) [inline nni/algorithms/compression/pytorch/pruning/sensitivity_pruner.py:164]: Only when both `sparsity` and `op_types/op_names` exists, this config is validate, right?

File: nni/algorithms/compression/pytorch/pruning/iterative_pruner.py (hunk 1 above)
Comment (J-shang) [reply nni/algorithms/compression/pytorch/pruning/iterative_pruner.py]: yes, seems a mistake, remove it.

File: nni/algorithms/compression/pytorch/pruning/sensitivity_pruner.py (hunk 2 above)
Comment (J-shang) [reply nni/algorithms/compression/pytorch/pruning/sensitivity_pruner.py]: yes, and you reminded me this way is not elegant, modify it.

File: nni/algorithms/compression/pytorch/pruning/dependency_aware_pruner.py [hunk 3]
```diff
@@ -82,17 +82,14 @@ def update_mask(self):
...
             Optional('sparsity'): And(float, lambda n: 0 < n < 1),
             Optional('op_types'): ['Conv2d'],
             Optional('op_names'): [str],
             Optional('exclude'): bool
         }], model, logger)
 
         schema.validate(config_list)
-        for config in config_list:
-            if 'exclude' not in config and 'sparsity' not in config:
-                raise SchemaError('Either sparisty or exclude must be specified!')
```
Comment (QuanluZhang) [inline nni/algorithms/compression/pytorch/pruning/dependency_aware_pruner.py:95]: why remove this validation?

**Review by QuanluZhang — APPROVED** [review]

**Comment by QuanluZhang** [comment]
@J-shang please fix pipeline errors

File: nni/algorithms/compression/pytorch/pruning/dependency_aware_pruner.py (hunk 3 above)
Comment (J-shang) [reply nni/algorithms/compression/pytorch/pruning/dependency_aware_pruner.py:95]: this validation move to `PrunerSchema` and `QuantizerSchema`

**Comment by J-shang** [comment]
> Do we have a documentation to describe the config interface of the pruner? Shall we give an example/unit test?

Yes, we have [doc](https://nni.readthedocs.io/en/stable/Compression/Tutorial.html?highlight=exclude#specify-the-configuration) and ut, and add an exclude usecase in the example.

**Review by QuanluZhang — APPROVED** [review]

**Review by zheng-ningxin — APPROVED** [review]

**Pull request merged**